packages/slate-react/src/components/content.js

```diff
@@ -93,6 +93,32 @@ class Content extends React.Component {
   }
 
   /**
+   * Should the component update?
+   *
+   * @param {Object} props
+   * @return {Boolean}
+   */
+
+  shouldComponentUpdate = (props) => {
+    // If the readOnly state has changed, we need to re-render so that
+    // the cursor will be added or removed again.
+    if (props.readOnly != this.props.readOnly) return true
+
+    // If the state has been changed natively, never re-render, or else we'll
+    // end up duplicating content.
+    if (props.state.isNative) return false
```

Comment:
This never appears to be true.  From what I can tell the change.state.isNative gets set to true, but this doesn't get updated in the props before it checks here.